src/main/java/io/debezium/connector/vitess/connection/VitessReplicationConnection.java

```diff
@@ -352,6 +370,9 @@ public static Vgtid defaultVgtid(VitessConnectorConfig config) {
         }
         else {
             if (config.getShard() == null || config.getShard().isEmpty()) {
+                if (config.getGtid().equals("")) {
```

Comment:
Although vitess doesn't support empty shard and empty gtid position, there is a workaround possible in debezium connector side.  We can get all the shards from vtgate and replace the empty shard param with all shards and continue (this was actually what was happening in VtGate code when you don't specify shard).  VitessConnector.getVitessShards() can get you all the shards.

Reply:
That sounds nice!
I was slanted toward that the connector side code was simpler and we would wait for the fix in Vitess.

But, given that [this use case is recommended at the document](https://debezium.io/documentation/reference/2.0/connectors/vitess.html#vitess-property-shard) and the current code has `VitessConnector.getVitessShards(),` I changed my mind that it's worth implementing the workaround now.
So, I'm getting to it.